From 9213244550335bcb2b8590a0d7d58ac74c932361 Mon Sep 17 00:00:00 2001
From: amodra <amodra@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 4 May 2016 13:23:58 +0000
Subject: [PATCH] [RS6000] Correct PIC_OFFSET_TABLE_REGNUM

Leaving this as r30 results in pic_offset_table_rtx of (reg 30)
for -m64, which is completely bogus.  Various rtl analysis predicate
functions treat pic_offset_table_rtx specially..

	* config/rs6000/rs6000.h (PIC_OFFSET_TABLE_REGNUM): Correct.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@235876 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog              | 1 +
 gcc/config/rs6000/rs6000.h | 5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/gcc/config/rs6000/rs6000.h b/gcc/config/rs6000/rs6000.h
index 12fa727..c89236a 100644
--- gcc-5-5.4.0.orig/src/gcc/config/rs6000/rs6000.h
+++ gcc-5-5.4.0/src/gcc/config/rs6000/rs6000.h
@@ -2058,7 +2058,10 @@ do {									     \
    to allocate such a register (if necessary).  */
 
 #define RS6000_PIC_OFFSET_TABLE_REGNUM 30
-#define PIC_OFFSET_TABLE_REGNUM (flag_pic ? RS6000_PIC_OFFSET_TABLE_REGNUM : INVALID_REGNUM)
+#define PIC_OFFSET_TABLE_REGNUM \
+  (TARGET_TOC ? TOC_REGISTER			\
+   : flag_pic ? RS6000_PIC_OFFSET_TABLE_REGNUM	\
+   : INVALID_REGNUM)
 
 #define TOC_REGISTER (TARGET_MINIMAL_TOC ? RS6000_PIC_OFFSET_TABLE_REGNUM : 2)
 
-- 
2.9.3

